From 07b6c8583c7f21fef466fb9b37d725f66d6594c4 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Thu, 20 Mar 2025 10:23:32 +0000
Subject: [PATCH 05/10] net: Add struct kernel_ethtool_ts_info

This commit accommodates commit 2111375b85ad("net: Add struct 
kernel_ethtool_ts_info") which renames ethtool_ts_info to 
kernel_ethtool_ts_info. This commit was merged in the v6.12 development
cycle and causes compilation failures with the v6.12 kernel, which are
listed below.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=2111375b85ad

Resolved compilation errors:

src/ice_ethtool.c: In function ice_get_ts_info:
src/ice_ethtool.c:5339:38: error: passing argument 2 of ethtool_op_get_ts_info
 from incompatible pointer type [-Werror=incompatible-pointer-types]
 5339 |   return ethtool_op_get_ts_info(dev, info);
      |                                      ^~~~
      |                                      |
      |                                      struct ethtool_ts_info *

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/ice_ethtool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ice_ethtool.c b/src/ice_ethtool.c
index e8aadaa..ceef806 100644
--- a/src/ice_ethtool.c
+++ b/src/ice_ethtool.c
@@ -5330,7 +5330,7 @@ ice_set_rxfh(struct net_device *netdev, const u32 *indir, const u8 *key)
 #endif /* ETHTOOL_GRSSH && ETHTOOL_SRSSH */
 
 static int
-ice_get_ts_info(struct net_device *dev, struct ethtool_ts_info *info)
+ice_get_ts_info(struct net_device *dev, struct kernel_ethtool_ts_info *info)
 {
 	struct ice_pf *pf = ice_netdev_to_pf(dev);
 
-- 
2.47.1

